File: Benchmarks/enterprise/java/jakarta/testcode/BenchmarkTest19958.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import jakarta.ws.rs.*;
import jakarta.ws.rs.core.Context;
import jakarta.ws.rs.core.MediaType;
import jakarta.ws.rs.core.Response;

@Path("/")
public class BenchmarkTest19958 {

    @GET
    @Path("/BenchmarkTest19958")
    @Produces(MediaType.APPLICATION_JSON)
    public Response BenchmarkTest19958(@HeaderParam("Host") String host, @Context HttpServletRequest request, @Context HttpServletResponse response) throws Exception {
        String hostValue = host != null ? host : "";
        String data = "[%s]".formatted(hostValue);
        if (!data.matches("^[a-zA-Z0-9_.-]+$")) { return Response.status(400).build(); }
        response.setContentType("application/json");
        return Response.ok("{\"echo\":\"" + data + "\"}", MediaType.APPLICATION_JSON).build();
    }
}

File: Benchmarks/enterprise/python/django/testcode/benchmark_test_19958.py
# SPDX-License-Identifier: Apache-2.0
from django.http import JsonResponse
import requests


class RequestPayload:
    def __init__(self, raw):
        self._raw = raw
    @property
    def value(self):
        return self._raw

def BenchmarkTest19958(request):
    user_id = request.GET.get('id', '')
    data = RequestPayload(user_id).value
    requests.get(str(data))
    return JsonResponse({"saved": True})
